98.md

```diff
@@ -0,0 +1,68 @@
+NIP-98
+======
+
+HTTP Auth
+-------------------------
+
+`draft` `optional` `author:kieran` `author:melvincarvalho`
+
+This NIP defines an ephemerial event used to authorize requests to HTTP servers using nostr events.
+
+This is useful for HTTP services which are build for Nostr and deal with Nostr user accounts.
+
+## Nostr event
+
+A `kind 27235` (In reference to [RFC 7235](https://www.rfc-editor.org/rfc/rfc7235)) event is used.
+
+The `content` SHOULD be empty.
+
+The following tags are defined as REQUIRED.
+
+* `u` - absolute URL
+* `method` - HTTP Request Method
+
+Example event:
+```json
+{
+    "id": "fe964e758903360f28d8424d092da8494ed207cba823110be3a57dfe4b578734",
+    "pubkey": "63fe6318dc58583cfe16810f86dd09e18bfd76aabc24a0081ce2856f330504ed",
+    "content": "",
+    "kind": 27235,
+    "created_at": 1682327852,
+    "tags": [
+        [
+            "u",
+            "https://api.snort.social/api/v1/n5sp/list"
+        ],
+        [
+            "method",
+            "GET"
+        ]
+    ],
+    "sig": "5ed9d8ec958bc854f997bdc24ac337d005af372324747efe4a00e24f4c30437ff4dd8308684bed467d9d6be3e5a517bb43b1732cc7d33949a3aaf86705c22184"
+}
+```
+
+Servers MUST perform the following checks in order to validate the event:
+1. The `kind` MUST be `27235`.
+2. The `created_at` MUST be within a reasonable time window (suggestion 60 seconds).
+3. The `url` tag MUST be exactly the same as the absolute request URL (including query parameters).
+4. The `method` tag MUST be the same HTTP method used for the requested resource.
+
+When the request contains a body (as in POST/PUT/PATCH methods) clients SHOULD include a SHA256 hash of the request body in a `payload` tag as hex (`["payload", "<sha256-hex>"]`), servers MAY check this to validate that the requested payload is authorized.
+
+If one of the checks was to fail the server SHOULD respond with a 401 Unauthorized response code.
```

Comment:
I believe this should be 403 if the Authorization header was actually sent.

Reply:
I would like to kindly request that we avoid introducing a breaking change at this advanced stage, especially since the implementations are already being used in real-world settings.

NAV-3 is implemented and will be using the original version. So, by introducing this change, we may unintentionally exclude Nostr users from personal storage on the web. This could put an increased load on our relays, which might not be the best outcome for all involved.  

Let's not forget our guiding principle stated in the NIPs which urges us to be mindful of avoiding breaking changes. Such changes not only have the potential to disrupt our current operations but can also discourage developers, perhaps even to the point where they might consider moving on.

More importantly, we have to remember the reputation of our project is at stake here. Breaking changes, especially ones that may not have clear and significant benefits, can potentially give the entire initiative a less than positive image.

Reply:
HTTP 401 Unauthorized: "I don't know who you are, please tell me before I can proceed"

HTTP 403 Forbidden: "I know who you are, but you still can't access this"

Reply:
I don't know what the constraints related to your project are, but if this is the correct usage of 401 that's fine, I'm not 100% clear on the nuances.